### PR TITLE
added ability to provide rp_project from command line

### PR DESCRIPTION
**Pull request by Formartha**
added ability to push --rp_project from command line

### CODE REVIEW

File: pytest_reportportal/plugin.py [hunk 1]
```diff
@@ -162,7 +164,8 @@ def pytest_configure(config):
         config._reportportal_configured = False
         return
 
-    project = config.getini('rp_project')
+    project = config.getini('rp_project')
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:167]:
```suggestion
    project = config.option.rp_project
    if not project:
        project = config.getini('rp_project')
```

File: pytest_reportportal/plugin.py [hunk 2]
```diff
@@ -72,7 +72,9 @@ def pytest_sessionstart(session):
     if is_master(session.config):
         try:
             session.config.py_test_service.init_service(
-                project=session.config.getini('rp_project'),
+                project=session.config.getini('rp_project')
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:75]:
```suggestion
                project=config.option.rp_project
```

**Review by iivanou — CHANGES_REQUESTED** [review]
Let's work with options from cmdline and ini file in the same manner.

**Comment by Formartha** [comment]
@iivanou - the commit I fixed related to PEP8 is great. I think you are incorrect with your comments and accidently I merged it. I'm submitting a new commit when the idea is that:
1. you will have ability to override pytest.ini (priority to cmd)
2. no PEP8 issues

Please comment here if you wish to fix something. the comment regarding "config" in session start scope is incorrect.

File: pytest_reportportal/plugin.py [hunk 3]
```diff
@@ -164,8 +164,9 @@ def pytest_configure(config):
         config._reportportal_configured = False
         return
 
-    project = config.getini('rp_project') if config.getini('rp_project') \
-        else config.option.rp_project
+    project = config.option.rp_project
+    if not project:
+        project = config.getini('rp_project')
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:169]:
```suggestion
        config.option.rp_project = project = config.getini('rp_project')
```
Once you do that, u'll be able to use session.config.option.rp_project in the pytest_sessionstart hook. pytest_configure hooks start before the pytest_sessionstart one.

File: pytest_reportportal/plugin.py [hunk 4]
```diff
@@ -72,9 +72,9 @@ def pytest_sessionstart(session):
     if is_master(session.config):
         try:
             session.config.py_test_service.init_service(
-                project=config.option.rp_project
-                if session.config.getini('rp_project')
-                else session.config.option.rp_project,
+                project=session.config.option.rp_project
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:75]: `project=session.config.option.rp_project` is enough if you apply the suggestion I left in the pytest_configure hook.

And, BTW, the current code may be simplified the following way:
`project=session.config.option.rp_project or session.config.getini('rp_project')`

**Review by iivanou — CHANGES_REQUESTED** [review]

**Comment by Formartha** [comment]
I'm not sure what's next. to be honest.. what do you want to do in order to merge it?

**Comment by iivanou** [comment]
> I'm not sure what's next. to be honest.. what do you want to do in order to merge it?

Make and commit requested changes.

File: pytest_reportportal/plugin.py (hunk 3 above)
Comment (Formartha) [reply pytest_reportportal/plugin.py:169]: that's what I'm seeing:
config.option.rp_project = project = config.getini('rp_project')

That's what I see. I don't understand what do you need me to do as it's not valid python code

Comment (iivanou) [reply pytest_reportportal/plugin.py:169]:
```
Python 3.6.5 (default, Apr 10 2018, 17:08:37) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> a = b = 300
>>> a
300
>>> b
300
>>> 
```

File: pytest_reportportal/plugin.py (hunk 4 above)
Comment (iivanou) [reply pytest_reportportal/plugin.py:75]: Still not fixed. Also, please update README with information about a new option.

File: pytest_reportportal/plugin.py [hunk 5]
```diff
@@ -72,7 +72,9 @@ def pytest_sessionstart(session):
     if is_master(session.config):
         try:
             session.config.py_test_service.init_service(
-                project=session.config.getini('rp_project'),
+                project=session.config.option.rp_project
+                if session.config.option.rp_project
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:76]:
```suggestion
```

File: pytest_reportportal/plugin.py [hunk 6]
```diff
@@ -72,7 +72,9 @@ def pytest_sessionstart(session):
...
+                else session.config.getini('rp_project'),
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:77]:
```suggestion
```

File: pytest_reportportal/plugin.py [hunk 7]
```diff
@@ -72,7 +72,9 @@ def pytest_sessionstart(session):
     if is_master(session.config):
         try:
             session.config.py_test_service.init_service(
-                project=session.config.getini('rp_project'),
+                project=session.config.option.rp_project
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:75]:
```suggestion
                project=session.config.option.rp_project,
```

File: pytest_reportportal/plugin.py [hunk 8]
```diff
@@ -162,7 +162,9 @@ def pytest_configure(config):
         config._reportportal_configured = False
         return
 
-    project = config.getini('rp_project')
+    project = config.option.rp_project
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:165]:
```suggestion
    project = config.option.rp_project or config.getini('rp_project')
```

File: pytest_reportportal/plugin.py [hunk 9]
```diff
@@ -162,7 +162,9 @@ def pytest_configure(config):
...
+    if not project:
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:166]:
```suggestion
```

File: pytest_reportportal/plugin.py [hunk 10]
```diff
@@ -162,7 +162,9 @@ def pytest_configure(config):
...
+        config.option.rp_project = project = config.getini('rp_project')
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:167]: config.option.rp_project = project

File: pytest_reportportal/plugin.py [hunk 11]
```diff
@@ -162,7 +162,8 @@ def pytest_configure(config):
         config._reportportal_configured = False
         return
 
-    project = config.getini('rp_project')
+    project = config.option.rp_project or config.getini('rp_project')
+        config.option.rp_project = project
```
Comment (iivanou) [inline pytest_reportportal/plugin.py:166]:
```suggestion
    config.option.rp_project = project
```

**Review by iivanou — APPROVED** [review]

**Pull request merged**